agent/src/main/java/com/teamscale/jacoco/agent/commit_resolution/git_properties/GitPropertiesLocatorUtils.java

```diff
@@ -85,22 +87,48 @@ public static String getRevisionFromGitProperties(
 		return null;
 	}
 
+	/**
+	 * VFS (Virtual File System) protocol is used by JBoss EAP and Wildfly. Example of an URL:
+	 * vfs:/content/helloworld.war/WEB-INF/classes
+	 */
 	private static Pair<File, Boolean> getVfsContentFolder(
 			URL jarOrClassFolderUrl) throws IOException, NoSuchMethodException, IllegalAccessException, InvocationTargetException {
-		// used by JBoss EAP and Wildfly. Example: vfs:/content/helloworld.war/WEB-INF/classes
-		Object virtualFile = jarOrClassFolderUrl.openConnection().getContent();
+		// we obtain the URL of a specific class file as input, e.g.,
+		// vfs:/content/helloworld.war/WEB-INF/classes
+		// Next, we try to extract the artefact URL from it, e.g., vfs:/content/helloworld.war
+		String artefactUrl = extractArtefactUrl(jarOrClassFolderUrl);
+		if (artefactUrl == null) {
+			return null;
+		}
+
+		Object virtualFile = new URL(artefactUrl).openConnection().getContent();
 		Class<?> virtualFileClass = virtualFile.getClass();
 		// obtain the physical location of the class file. It is created on demand in <jboss-installation-dir>/standalone/tmp/vfs
 		Method getPhysicalFileMethod = virtualFileClass.getMethod("getPhysicalFile");
 		File file = (File) getPhysicalFileMethod.invoke(virtualFile);
-		if (file.getParentFile() == null || file.getParentFile().getParentFile() == null) {
+		return Pair.createPair(file, false);
+	}
+
+	private static String extractArtefactUrl(URL jarOrClassFolderUrl) {
+		String url = jarOrClassFolderUrl.getPath().toLowerCase();
+		String[] pathSegments = url.split("/");
+		StringBuilder artefactUrlBuilder = new StringBuilder("vfs:");
+		int segmentIdx = 0;
+		String segment = pathSegments[segmentIdx];
+		while (segmentIdx < pathSegments.length && !org.conqat.lib.commons.string.StringUtils.endsWithOneOf(
+				segment, ".jar", ".war", ".ear", ".aar")) {
+			artefactUrlBuilder.append(segment);
+			artefactUrlBuilder.append("/");
+			if (segmentIdx + 1 < pathSegments.length) {
+				segment = pathSegments[segmentIdx + 1];
+			}
+			segmentIdx += 1;
+		}
+		if (segmentIdx == pathSegments.length) {
 			return null;
```

Comment:
Do we really want to return null here? I would have expected the full path. Not sure if that is possible in JBoss, but I could image that you can besides jar's war's etc. also add folders of class files which would presumably end up as vfs:/content/path/to/my/folder

Reply:
What do you mean by "add folders of class files"? You mean deploy class files to the application server directly, without a jar/war artefact? I don't know if something like this is possible. However, if I return the full path where the class is residing, then further git.properties search will only be done relative to this path, is that fine?

Reply:
The alternative would be to skip looking in that location entirely, right?